src/AaveV3StrategicAssets_20230606/VeTokenManager.sol

```diff
@@ -0,0 +1,224 @@
+  event SellBoost(
+    uint256 pricePerVote,
+    uint64 duration,
+    uint64 expiryTime,
+    uint16 minPerc,
+    uint16 maxPerc,
+    bool useAdvicePrice
+  );
+  event SetLockDuration(uint256 newDuration);
+  event SetSpaceId(bytes32 id);
+  event UnlockVEBAL(uint256 tokensUnlocked);
+
+  error MaxFeeExceeded();
+  error NoTokensToLockOrRelock();
+
+  IDelegateRegistry public constant DELEGATE_REGISTRY =
+    IDelegateRegistry(0x469788fE6E9E9681C6ebF3bF78e7Fd26Fc015446);
+
+  address public constant BAL = 0xba100000625a3754423978a60c9317c58a424e3D;
+  address public constant B_80BAL_20WETH = 0x5c6Ee304399DBdB9C8Ef030aB642B10820DB8F56;
+  address public constant VE_BAL = 0xC128a9954e6c874eA3d62ce62B468bA073093F25;
+  address public constant WARDEN_VE_BAL = 0x42227bc7D65511a357c43993883c7cef53B25de9;
+
+  /// @notice the snapshot delegate for the deposit
+  address public delegate;
+  /// @notice the keccak encoded spaceId of the snapshot space
+  bytes32 public spaceIdBalancer;
+  /// @notice The lock duration of veToken
+  uint256 public lockDurationVEBAL;
+
+  /// @notice Duration in weeks (ie: 1 for 1 Week)
+  /// @param delegator Address of user to buy boost from
+  /// @param receiver Address receiving the boost
+  /// @param amount The amount of boost to purchase
+  /// @param durationWeeks The number of weeks to buy boost for
+  /// @param maxFeeOffered The maximum fee the buyer is willing to pay
+  function buyBoost(
+    address delegator,
+    address receiver,
+    uint256 amount,
+    uint256 durationWeeks,
+    uint256 maxFeeOffered
+  ) external onlyOwnerOrGuardian {
+    uint256 maxFee = IWardenBoost(WARDEN_VE_BAL).estimateFees(delegator, amount, durationWeeks);
+
+    if (maxFee > maxFeeOffered) revert MaxFeeExceeded();
+
+    IERC20(BAL).forceApprove(WARDEN_VE_BAL, maxFee);
+    IWardenBoost(WARDEN_VE_BAL).buyDelegationBoost(
+      delegator,
+      receiver,
+      amount,
+      durationWeeks,
+      maxFee
+    );
+
+    emit BuyBoost(delegator, receiver, amount, durationWeeks);
+  }
+
+  /// @notice Sells boost in order to accrue rewards
+  /// @dev Registers a new user, creates a BoostOffer with the given parameters
+  /// @param pricePerVote Price of 1 vote per second (in wei)
+  /// @param maxDurationWeeks Maximum duration (in weeks) that a Boost can last when taken from this Offer
+  /// @param expiryTime Timestamp when this Offer is no longer valid
+  /// @param minPerc Minimum percent of users voting token balance to buy for a Boost (in BPS)
+  /// @param maxPerc Maximum percent of users total voting token balance available to delegate (in BPS)
+  /// @param useAdvicePrice True to use the advice Price instead of the given pricePerVote
+  function sellBoost(
+    uint256 pricePerVote,
+    uint64 maxDurationWeeks,
+    uint64 expiryTime,
+    uint16 minPerc,
+    uint16 maxPerc,
+    bool useAdvicePrice
+  ) external onlyOwnerOrGuardian {
+    IERC20(IWardenBoost(WARDEN_VE_BAL).delegationBoost()).forceApprove(
```

Comment:
[info] Maybe a bit inefficient to call this on every sell, but this doesn't seem to be a very high velocity function, so I don't really see an issue. Just flagging.